nixos/modules/services/games/factorio.nix

```diff
@@ -37,7 +36,7 @@ let
     only_admins_can_pause_the_game = true;
     autosave_only_on_server = true;
     admins = [];
-  };
+  } // cfg.extraSettings;
```

Comment:
instead of adding an option to override all of the other configuration, why don't you extend the current options to include admins and others.

Also, you should probably be using something like mkMerge to ensure that the keys are gracefully composed.

Reply:
Extending the current options seems like a good idea. I didn't want to bloat the options too much, but that would definately help with documentation. I'll update this PR and add an option for admins.

Reply:
I wanted to create a factorio server with `services.factorio` and everything worked straight away. However, I can't make myself an admin, not even when I delete the state directory and restart the service with `extraConfig = { admins = [ "username" ] }` in place. Is `extraConfig` currently broken or am I doing something wrong? As far as I see it's generally not possible to connect a TTY to the server and run `/promote <username>`.

Reply:
I had this issue as well, I'll look into it. But probably not til after the 20.09 release

Reply:
Great, thank you!

Reply:
that being said, do you mind making an issue on github? better way to track than a comment, and please cc me